### DISCUSSION TITLE
NET6 dll in .Net Framework 4.8 project

### DISCUSSION THREAD

**Discussion by Wlad57** [lepoco/wpfui#326] [Q&A]
Hi, I have one question regarding compatibility with older frameworks like nf48.

Since the library is written in NET6 (C#10) how can we be sure that it will work correctly in net framework 4.8 (C#7)? Does the compiler downgrade the language at build time or something else?

This is probably a simple question but I couldn't find an answer online :)

**Comment by pomianowski** [comment]
In .NET this is done in such a way that when building a library, separate DLLs are created for each version. And if you look in the NuGet archive, you'll notice that there is a separate folder for each version.
![image](https://user-images.githubusercontent.com/13592821/184632565-1d4ca92f-2361-4fff-905d-7988249837fe.png)


If any library or functionality is not supported, preprocessor directives are used
![image](https://user-images.githubusercontent.com/13592821/184632897-8c74faac-b67f-46b8-b7cc-ec21a6009062.png)

 - C# preprocessor directives
 https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives
- NuGet Support multiple .NET versions
 https://docs.microsoft.com/en-us/nuget/create-packages/supporting-multiple-target-frameworks
